packages/medusa/src/strategies/batch-jobs/product/import.ts

```diff
@@ -270,13 +294,31 @@ class ProductImportStrategy extends AbstractBatchJobStrategy {
    * @param batchJobId - An id of a batch job that is being processed.
    */
   async processJob(batchJobId: string): Promise<void> {
-    return await this.atomicPhase_(async () => {
-      await this.createProducts(batchJobId)
-      await this.updateProducts(batchJobId)
-      await this.createVariants(batchJobId)
-      await this.updateVariants(batchJobId)
+    return await this.atomicPhase_(async (manager) => {
+      const batchJob = await this.batchJobService_
+        .withTransaction(manager)
+        .retrieve(batchJobId)
+      const operations = batchJob.result.operations as {
+        [K in keyof typeof OperationType]: number
+      }
+
+      if (operations[OperationType.ProductCreate]) {
```

Comment:
question: is all this necessary including `operationsCounts` changes in preprocessing? All this could be replaced in `downloadImportOpsFile` by catching if the file doesn't exist and returning `[]` in that case

Reply:
I think that since we have the data that are already computed from the method used in the pre processing, we can take advantage of it to not start operations that does not need to be done. It can avoid an expected side effect too 😅

why do you think we should try catch instead?

Reply:
This still doesn't guarantee that error won't be thrown, if we want to be sure, we should check whether the file exists in the bucket or not. That way the implementation could be simpler because we have a check in a single place.

The idea before was that every (create/update...) method was responsible for figuring out if there are import instructions to be processed by this method and that logic was encapsulated within these methods.

Reply:
Wdyt about that approach https://github.com/medusajs/medusa/pull/2150/commits/08c2639073b2e6eaa4c4fd4726a685a31e8e46fe ?

Reply:
Looks good 👍